admin_guide/cluster_autoscaler.adoc

```diff
@@ -0,0 +1,527 @@
+[[admin-guide-cluster-autoscaler]]
+= Managing the Cluster Auto Scaler
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[managing-autoscaler-overview]]
+== Overview
+
+This topic describes setting up the auto-scaler on your
+{product-title} cluster. The auto-scaler's remit is to ensure your
+pods have somewhere to run, and to also ensure there are no unneeded
+nodes when demand drops.
+
+Note: this feature is currently in tech-preview and the this guide
+only applies to running the auto-scaler on AWS.
+
+=== Auto Scale Groups
+
+An Auto Scale Group (ASG) is a logical representation of a set of
+machines. An ASG is configured with a minimum number of instances to
+run, a maximum number of instances that can be run and also the
+desired number of instances to run. An ASG starts by launching enough
+instances to meet its desired capacity. ASGs may start with a minimum
+of zero instances.
+
+=== Launch Configurations
+
+A Launch Configuration (LC) is a template that an ASG uses to launch
+instances; you specify information such as the ID of the Amazon
+Machine Image (AMI), the instance type (e.g., m4.large), a key pair,
+one or more security groups, subnets, etc.
+
+=== OpenShift "golden" Images
+
+When the ASG provisions a new instance the image that is run must have
+OpenShift already prepared so that it can be both automatically
+bootstrapped and enrolled within the cluster without manual
+intervention.
+
+=== Operation
+
+The auto-scaler periodically checks (e.g., every 30 seconds) to see
+how many pods are pending node allocation. If there are nodes pending
+allocation and the auto-scaler has not met its maximum capacity then
+new nodes are continuously provisioned to accommodate the current
+demand. And, equally, when demand drops the auto-scaler will remove
+nodes, thereby saving you money.
+
+== Installation
+
+There are currently three ways to create a cluster that has auto
+scaling capabilities.
+
+1. The first way is to install the auto-scaler automatically using the
+   AWS ansible playbooks when creating a cluster from scratch. This
+   method is a complete hands-off automated install that will
+   provisioning the cluster nodes and install OpenShift, that includes
+   all configuration of the auto-scaler.
+
+2. The second way is to retroactively add the auto-scaler to a cluster
+   that was previously setup using method (1).
+
+3. The third way is to manually create the ASG and LC, then manually
+   deploy the auto-scaler components using the
+   openshift-cluster-autoscaler ansible playbooks.
```

Comment:
In what cases is it better to run these manually instead of running the playbooks?

Reply:
Do you have to already have a cluster in AWS to complete this method?

Reply:
> In what cases is it better to run these manually instead of running the playbooks?

If you already have a cluster that was setup up through the `.../playbooks/aws/...' then it it better to continue with the AWS playbooks. If you have an existing cluster that was a) provisioned manually and b) the `deploy_cluster.yaml` playbook was run then the third was is your only choice.

Reply:
> Do you have to already have a cluster in AWS to complete this method?

Yes.